### PR TITLE
Replace basically everything

### LINKED ISSUE #8: Update list of packages
Forward mode:
- [ForwardDiff](https://github.com/JuliaDiff/ForwardDiff.jl)
- [ForwardDiff2](https://github.com/YingboMa//ForwardDiff2.jl)

Reverse Mode:
- [Nabla](https://github.com/invenia/Nabla.jl/)
- [Tracker](https://github.com/FluxML/Tracker.jl)
- [Yota](https://github.com/dfdx/Yota.jl)
- [Zygote](https://github.com/FluxML/Zygote.jl)
- [ReverseDiff](https://github.com/JuliaDiff/ReverseDiff.jl)
- [AutoGrad.jl](https://github.com/denizyuret/AutoGrad.jl)
- [NiLang](https://github.com/GiggleLiu/NiLang.jl) (arguably not reverse mode)

Symbolic:
- [ModelingToolKit](https://github.com/JuliaDiffEq/ModelingToolkit.jl)
- [XGrad.jl](https://github.com/dfdx/XGrad.jl)

Finite Differencing:

- [Calculus](https://github.com/JuliaMath/Calculus.jl) (please stop)
- [FiniteDifferences](https://github.com/JuliaDiff/FiniteDifferences.jl)
- [FiniteDiff](https://github.com/JuliaDiff/FiniteDiff.jl)

Rulesets:
- [ChainRules](https://github.com/JuliaDiff/ChainRulesCore.jl)
- [DiffRules](https://github.com/JuliaDiff/ChainRulesCore.jl) (soon to stop)

Sparsity Detection
- [SparseDiffTools](https://github.com/JuliaDiff/SparseDiffTools.jl)

### PR DESCRIPTION
**Pull request by oxinabox**
Closes #8 and #9

Rewrote in Franklin.
So could have markdown.

Deleted the middle content and wrote a new big list that was up to date.
Possibly should also rewrite the introduction, since AD is no longer surprising to many people? idk.
Can be a follow-up PR

Most the files are the auto-generated stuff from Franklin.
the `index.md` has the actual content

image
image

### CODE REVIEW

File: JuliaDiff/index.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+- [ModelingToolKit.jl](https://github.com/JuliaDiffEq/ModelingToolkit.jl): A pure Julia computer algebra system. While its docs focus on some particular domain use-case it is a fully general purpose system.
+
+### Exotic
+- [TaylorSeries.jl](https://github.com/JuliaDiff/TaylorSeries.jl) computes polynomial expansions; which is the generalization of forward-mode AD to n-th order derivatives.
+- [NiLang.jl](https://github.com/GiggleLiu/NiLang.jl): Reversible computing DSL, where everything is differentiable, by reversing.
+
+### Finite Differencing
+(yes, we said at the start to stop approximating deriviatives, but these packages are faster and more accurate than you would expect finite differencing to every achieve.)
+
+- [FiniteDifferences.jl](https://github.com/JuliaDiff/FiniteDifferences.jl): High-accuracy finite differencing with support for almost any type (not just the usual arrays and numbers): [FiniteDiff](https://github.com/JuliaDiff/FiniteDiff.jl): High-accuracy finite differencing with support for efficient calculation of spares Jacobians via coloring vectors.
```
Comment (ChrisRackauckas) [inline JuliaDiff/index.md:61]:
```suggestion
- [FiniteDifferences.jl](https://github.com/JuliaDiff/FiniteDifferences.jl): High-accuracy finite differencing with support for almost any type (not just the usual arrays and numbers)
- [FiniteDiff](https://github.com/JuliaDiff/FiniteDiff.jl): High-accuracy finite differencing with support for efficient calculation of spares Jacobians via coloring vectors.
```

File: JuliaDiff/Project.toml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[deps]
+Franklin = "713c75ef-9fc9-4b05-94a9-213340da978e"
+NodeJS = "2bd173c7-0d6d-553b-b6af-13a54713934c"
```
Comment (matbesancon) [inline JuliaDiff/Project.toml:3]:
```suggestion
NodeJS = "2bd173c7-0d6d-553b-b6af-13a54713934c"

```

just new line at the end (ignore if file is generated)

File: JuliaDiff/index.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+
+#### What can we do instead?
+One option is to explicitly write down a function which computes the exact derivatives by using the rules that we know from calculus. However, this quickly becomes an error-prone and tedious exercise. **There is another way!** The field of [automatic differentiation]("https://en.wikipedia.org/wiki/Automatic_differentiation") provides methods for automatically computing _exact_ derivatives (up to floating-point error) given only the function $ f $ itself. Some methods use many fewer evaluations of $ f $ than would be required when using finite differences. In the best case, **the exact gradient of $ f $ can be evaluated for the cost of $ O(1) $ evaluations of $ f $ itself**.  The caveat is that $ f $ cannot be considered a black box; instead, we require either access to the source code of $ f $ or a way to plug in a special type of number using operator overloading.
```
Comment (matbesancon) [inline JuliaDiff/index.md:18]:
```suggestion
One option is to explicitly write down a function which computes the exact derivatives by using the rules that we know from calculus. However, this quickly becomes an error-prone and tedious exercise. **There is another way!** The field of [automatic differentiation](https://en.wikipedia.org/wiki/Automatic_differentiation) provides methods for automatically computing _exact_ derivatives (up to floating-point error) given only the function $ f $ itself. Some methods use many fewer evaluations of $ f $ than would be required when using finite differences. In the best case, **the exact gradient of $ f $ can be evaluated for the cost of $ O(1) $ evaluations of $ f $ itself**.  The caveat is that $ f $ cannot be considered a black box; instead, we require either access to the source code of $ f $ or a way to plug in a special type of number using operator overloading.
```

I'm not sure the quotes change anything, but I've never seen them in MD reference

File: JuliaDiff/index.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+### Exotic
+- [TaylorSeries.jl](https://github.com/JuliaDiff/TaylorSeries.jl) computes polynomial expansions; which is the generalization of forward-mode AD to n-th order derivatives.
+- [NiLang.jl](https://github.com/GiggleLiu/NiLang.jl): Reversible computing DSL, where everything is differentiable, by reversing.
+
+### Finite Differencing
+(yes, we said at the start to stop approximating deriviatives, but these packages are faster and more accurate than you would expect finite differencing to every achieve.)
+
+- [FiniteDifferences.jl](https://github.com/JuliaDiff/FiniteDifferences.jl): High-accuracy finite differencing with support for almost any type (not just the usual arrays and numbers)
+- [FiniteDiff](https://github.com/JuliaDiff/FiniteDiff.jl): High-accuracy finite differencing with support for efficient calculation of spares Jacobians via coloring vectors.
```
Comment (nickrobinson251) [inline JuliaDiff/index.md:62]:
```suggestion
- [FiniteDiff.jl](https://github.com/JuliaDiff/FiniteDiff.jl): High-accuracy finite differencing with support for efficient calculation of spares Jacobians via coloring vectors.
```

File: JuliaDiff/index.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+- [Calculus.jl](https://github.com/JuliaMath/Calculus.jl): Largely deprecated, legacy package. New users should look to FiniteDifferences and FiniteDiff instead.
```
Comment (nickrobinson251) [inline JuliaDiff/index.md:63]:
```suggestion
- [Calculus.jl](https://github.com/JuliaMath/Calculus.jl): Largely deprecated, legacy package. New users should look to FiniteDifferences.jl and FiniteDiff.jl instead.
```

File: JuliaDiff/index.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+### Rulesets:
```
Comment (nickrobinson251) [inline JuliaDiff/index.md:65]:
```suggestion
### Rulesets
Packages providing collections of derivatives of functions which can be used in AD packages.
```

File: JuliaDiff/index.md [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+- [ChainRules.jl](https://www.juliadiff.org/ChainRulesCore.jl/stable/): extensible AD independent rule-set.
```
Comment (nickrobinson251) [inline JuliaDiff/index.md:66]: Maybe some way to distinguish ChainRules-ecosystem from just ChainRules.jl?

```suggestion
- [ChainRules](https://www.juliadiff.org/ChainRulesCore.jl/stable/): Extensible, AD-independent rules.
```

File: JuliaDiff/index.md [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+  - [ChainRulesCore.jl](https://github.com/JuliaDiff/ChainRulesCore.jl): Core API for user to extend to add rules to their package.
+  - [ChainRules.jl](https://github.com/JuliaDiff/ChainRules.jl/): the rules for Base and other standard libraries.
```
Comment (nickrobinson251) [inline JuliaDiff/index.md:68]:
```suggestion
  - [ChainRules.jl](https://github.com/JuliaDiff/ChainRules.jl/): Rules for Julia Base and standard libraries.
```

File: JuliaDiff/index.md [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+  - [ChainRulesTestUtils.jl](https://github.com/JuliaDiff/ChainRulesTestUtils.jl/): testing tools to check rules defined with ChainRulesCore.
```
Comment (nickrobinson251) [inline JuliaDiff/index.md:69]:
```suggestion
  - [ChainRulesTestUtils.jl](https://github.com/JuliaDiff/ChainRulesTestUtils.jl/): Tools for testing rules defined with ChainRulesCore.jl.
```

File: JuliaDiff/index.md [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+- [DiffRules.jl](https://github.com/JuliaDiff/ChainRulesCore.jl): Original multi-AD system rule-set. Largely deprecated in-favor of ChainRules.
```
Comment (nickrobinson251) [inline JuliaDiff/index.md:70]:
```suggestion
- [DiffRules.jl](https://github.com/JuliaDiff/ChainRulesCore.jl): An earlier set of AD-independent rules. Largely deprecated in favor of the more extensible ChainRules.jl.
```

File: JuliaDiff/index.md [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+- [ZygoteRules.jl](https://github.com/FluxML/ZygoteRules.jl/blob/master/src/ZygoteRules.jl): for adding rules to Zygote. Largely deprecated in-favour of ChainRules.
```
Comment (nickrobinson251) [inline JuliaDiff/index.md:71]:
```suggestion
- [ZygoteRules.jl](https://github.com/FluxML/ZygoteRules.jl/blob/master/src/ZygoteRules.jl): For defining Zygote.jl specific rules. Largely deprecated in favour of the AD-independent ChainRules.jl.
```

File: JuliaDiff/index.md [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+
+### Sparsity
+- [SparsityDetection.jl](https://github.com/SciML/SparsityDetection.jl): automatic Jacobian and Hessian sparsity pattern detection.
```
Comment (nickrobinson251) [inline JuliaDiff/index.md:74]:
```suggestion
- [SparsityDetection.jl](https://github.com/SciML/SparsityDetection.jl): Automatic Jacobian and Hessian sparsity pattern detection.
```

File: JuliaDiff/index.md [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+- [SparseDiffTools.jl](https://github.com/JuliaDiff/SparseDiffTools.jl): exploiting sparsity to speed up FiniteDiff and ForwardDiff; as well as other algorithms.
```
Comment (nickrobinson251) [inline JuliaDiff/index.md:75]:
```suggestion
- [SparseDiffTools.jl](https://github.com/JuliaDiff/SparseDiffTools.jl): Exploiting sparsity to speed up FiniteDiff.jl and ForwardDiff.jl, as well as other algorithms.
```

File: JuliaDiff/index.md [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+
+
+## Links for discussion and more information
+Discussions on JuliaDiff and its uses may be directed to the [Julia Discourse forum](https://discourse.julialang.org/)
+The [autodiff.org](http://www.autodiff.org/) site serves as a portal for the academic community, though it is often out-of-date.
+The ChainRules project maintains a [list of recommend-reading/watching](https://www.juliadiff.org/ChainRulesCore.jl/stable/FAQ.html#Where-can-I-learn-more-about-AD-?) for those after more information.
```
Comment (nickrobinson251) [inline JuliaDiff/index.md:81]:
```suggestion
The ChainRules project maintains a [list of recommend reading/watching](https://www.juliadiff.org/ChainRulesCore.jl/stable/FAQ.html#Where-can-I-learn-more-about-AD-?) for those after more information.
```

File: JuliaDiff/index.md [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+Finally, automatic differentiation techniques have been implemented in a variety of languages.
+If you would prefer not to use Julia, see the [wikipedia page](http://en.wikipedia.org/wiki/Automatic_differentiation") for a comprehensive list of available packages.
```
Comment (nickrobinson251) [inline JuliaDiff/index.md:83]:
```suggestion
If you would prefer not to use Julia, see the [wikipedia page](http://en.wikipedia.org/wiki/Automatic_differentiation) for a comprehensive list of available packages.
```

**Review by nickrobinson251 — COMMENTED** [review]
Thanks for doing this! Was much needed :D

Left a load of small suggestions to make the text more consistent.

**Comment by nickrobinson251** [comment]
also, looking at the JuliaDiff org, should we mention DualNumbers? and should we just archive Capstan?

File: JuliaDiff/index.md (hunk 7 above)
Comment (oxinabox) [reply JuliaDiff/index.md:66]: Ah yes I mean to do that but it got caught in a global find-replace when I added jl to everything

**Comment by oxinabox** [comment]
> also, looking at the JuliaDiff org, should we mention DualNumbers?

Maybe?
It's pedagogical, rather than useful.
So we could add a learning resources section here and include it and all the stuff from ChainRules's links.

> and should we just archive Capstan?

Maybe we should have a "of historical note" section.
And put that and other things in it.
For people who find old docs etc.
After that I would be happy archiving it as well as HyperDuelNumbers.
Since they would still be discoverable.


Both can be addressed in future PRs

**Comment by mlubin** [comment]
I don't mind if you rewrite the intro.

**Comment by oxinabox** [comment]
> I don't mind if you rewrite the intro.

Will do in a follow up PR

**Comment by oxinabox** [comment]
@mlubin can I have your blessing to merge this?
and then I will probably merge a bunch of smaller changes to make sure the site builds

**Review by mlubin — APPROVED** [review]

**Pull request merged**